docs/OpenSSLSupport.md

```diff
@@ -0,0 +1,197 @@
+# Open Enclave Support for OpenSSL
+
+The OpenSSL on OE is configured with following options
+- no-hw
+- no-shared
+- no-threads
+- no-dso
+- no-ssl2
+- no-ss3
+- --with-rand-seed=none
+
+OE also explicitly disables the following features
+- AFALGENG (AF_ALG hardware engine)
+  - Require special hardware.
+- GOST (Russian GOST crypto engine)
+  - Require dynamic loading.
+- SECURE_MEMORY
+  - Require mman features (e.g., madvise, mlock, mprotect) that are not supported inside an enclave.
+- UI_CONSOLE
+  - Used by the openssl command-line tool that is not required by OE.
+
+In addition, OpenSSL by default disables the following algorithms/features
+- MD2
+- RC5
+- EC_NISTP_64_GCC_128
+- EGD (Entropy Gathering Daemon)
+- Heartbeats extension
+- SCTP (Stream Control Transimission Protocol) protocol
+
+*NOTE*: The current support is still experimental and for SGX only. To use OpenSSL libraries, developers should specify
+`-DBUILD_OPENSSL=ON` cmake option when building the OE SDK.
+
+# How to use RAND APIs
+
+Currently, the default RAND method used by RAND APIs is not supported by OE. More specifically,
+the default OpenSSL RAND method relies on the `rdtsc` instruction, which is not supported by SGXv1 enclaves.
```

Comment:
For now it's mainly because of the former (i.e., SGX1 does not support it). As @CodeMonkeyLeet suggested, we will discuss this part (including the security concerns you pointed out) in a separated document before moving OpenSSL out of experimental features.

Reply:
Ok. Is there some issue that is capturing all the follow up items?

Reply:
@CodeMonkeyLeet's summary of follow-up should cover this already. I can open an issue to track if needed.